pkg/operator/sapinstancestart_v1.go

```diff
@@ -0,0 +1,290 @@
+//	instance_number (required): String with the instance number of the instance to start
+//  timeout: Timeout in seconds to wait until the instance is started
+//
+// # Execution Phases
+//
+// - PLAN:
+//   The operator gets the instance current processes and stores the state.
+//
+// - COMMIT:
+//   If the SAP instances is not already started, it is started using the sapcontrol Start command.
+//
+// - VERIFY:
+//   Verify if the SAP instance is started.
+//
+// - ROLLBACK:
+//   If an error occurs during the COMMIT or VERIFY phase, the instance is stopped back again.
+
+type SAPInstanceStart struct {
+	baseOperator
+	parsedArguments     *sapStateChangeArguments
+	sapControlConnector sapcontrol.SAPControlConnector
+	interval            time.Duration
+	startDelay          time.Duration
+}
+
+func WithCustomStartSapcontrol(sapControlConnector sapcontrol.SAPControlConnector) SAPInstanceStartOption {
+	return func(o *SAPInstanceStart) {
+		o.sapControlConnector = sapControlConnector
+	}
+}
+
+func WithCustomStartInterval(interval time.Duration) SAPInstanceStartOption {
+	return func(o *SAPInstanceStart) {
+		o.interval = interval
+	}
+}
+
+func WithCustomStartInitialDelay(startDelay time.Duration) SAPInstanceStartOption {
+	return func(o *SAPInstanceStart) {
+		o.startDelay = startDelay
+	}
+}
+
+func NewSAPInstanceStart(
+	arguments OperatorArguments,
+	operationID string,
+	options OperatorOptions[SAPInstanceStart],
+) *Executor {
+	sapInstanceStart := &SAPInstanceStart{
+		baseOperator: newBaseOperator(operationID, arguments, options.BaseOperatorOptions...),
+		interval:     defaultSapInstanceStateInterval,
+		startDelay:   defaultStartDelay,
+	}
+
+	for _, opt := range options.OperatorOptions {
+		opt(sapInstanceStart)
+	}
+
+	return &Executor{
+		phaser:      sapInstanceStart,
+		operationID: operationID,
+	}
+}
+
+func (s *SAPInstanceStart) plan(ctx context.Context) error {
+	opArguments, err := parseSAPStateChangeArguments(s.arguments)
+	if err != nil {
+		return err
+	}
+	s.parsedArguments = opArguments
+
+	// Use custom sapControlConnector or create a new one based on the instance_number argument
+	if s.sapControlConnector == nil {
+		s.sapControlConnector = sapcontrol.NewSAPControlConnector(s.parsedArguments.instNumber)
+	}
+
+	started, err := allProcessesInState(ctx, s.sapControlConnector, sapcontrol.STATECOLORSAPControlGREEN)
+	if err != nil {
+		return fmt.Errorf("error checking processes state: %w", err)
+	}
+
+	s.resources[beforeDiffField] = started
+
+	return nil
+}
+
+func (s *SAPInstanceStart) commit(ctx context.Context) error {
+	if s.resources[beforeDiffField] == true {
+		s.logger.Info("instance already started, skipping operation")
+		return nil
+	}
+
+	request := new(sapcontrol.Start)
+	_, err := s.sapControlConnector.StartContext(ctx, request)
+	if err != nil {
+		return fmt.Errorf("error starting instance: %w", err)
+	}
+
+	// need to wait until the processes start properly
+	err = sleepContext(ctx, s.startDelay)
```

Comment:
Do I get it right that this `sleepContext` here will only sleep for `s.startDelay`(1 second in this case)? 

There is no context done in this usage, right? while the context could be done in `sleepContext` usage inside the following `waitUntilSapInstanceState`.

Reply:
> Do I get it right that this sleepContext here will only sleep for s.startDelay(1 second in this case)?

Yes, it will sleep for 1 second, unless the context is cancelled before this `startDelay` which in this case, it will return inmediately.

>There is no context done in this usage, right? while the context could be done in sleepContext usage inside the following waitUntilSapInstanceState.

No, the context is checked, and it is used to cancel the sleep if the context is cancelled. We cannot wait until the usage of `waitUntilSap...`, because that function doesn't have an initial sleep before entering the loop